### PR TITLE
[RLlib] Examples folder cleanup: ModelV2 -> RLModule wrapper for migrating to new API stack.

### PR DESCRIPTION
**Pull request by sven1977**
Adds example script and RLModule example class for easy ModelV2 -> RLModule migration to new API stack using an old API stack Policy checkpoint.

Up next: Enhance the wrapper to work with a `from_config` option instead of the currently needed checkpoint.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/core/rl_module/apis/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from ray.rllib.core.rl_module.apis.target_network_api import TargetNetworkAPI
+from ray.rllib.core.rl_module.apis.value_function_api import ValueFunctionAPI
+
+
+__all__ = [
```
Comment (simonsays1980) [inline rllib/core/rl_module/apis/__init__.py:5]: Nice!!

File: rllib/examples/rl_modules/classes/modelv2_to_rlm.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        self._model_v2 = policy.model
+
+        # Translate the action dist classes from the old API stack to the new.
+        self._action_dist_class = self._translate_dist_class(policy.dist_class)
+
+        # Erase the torch policy from memory, so it can be garbage collected.
+        del policy
+
+    def _forward_inference(self, batch: Dict[str, Any], **kwargs) -> Dict[str, Any]:
+        nn_output, state_out = self._model_v2(batch)
```
Comment (simonsays1980) [inline rllib/examples/rl_modules/classes/modelv2_to_rlm.py:55]: Nice!!! So simple. I wonder if this works fine with any more complicated setup like LSTM or MARWIL

File: rllib/tuned_examples/bc/cartpole_recording.py [hunk 3]
```diff
@@ -31,11 +31,9 @@
         }
     )
     .training(
-        gamma=0.99,
```
Comment (simonsays1980) [inline rllib/tuned_examples/bc/cartpole_recording.py:34]: Ah thanks. Forgot these ...

File: rllib/tuned_examples/ppo/multi_agent_stateless_cartpole_ppo.py [hunk 4]
```diff
@@ -37,10 +37,8 @@
     )
     .training(
         lr=0.0003 * ((args.num_gpus or 1) ** 0.5),
-        gamma=0.99,
```
Comment (simonsays1980) [inline rllib/tuned_examples/ppo/multi_agent_stateless_cartpole_ppo.py:40]: Removing b/c they are default?

File: rllib/tuned_examples/ppo/stateless_cartpole_ppo.py [hunk 5]
```diff
@@ -29,10 +29,8 @@
     )
     .training(
         lr=0.0003 * ((args.num_gpus or 1) ** 0.5),
-        gamma=0.99,
```
Comment (simonsays1980) [inline rllib/tuned_examples/ppo/stateless_cartpole_ppo.py:32]: Same here ..

**Review by simonsays1980 — COMMENTED** [review]
LGTM. Great PR. This enables hopefully more users to move from old to new stack!

File: rllib/examples/rl_modules/classes/modelv2_to_rlm.py (hunk 2 above)
Comment (sven1977) [reply rllib/examples/rl_modules/classes/modelv2_to_rlm.py:55]: Probably not :| but it's just a start ...

File: rllib/tuned_examples/ppo/multi_agent_stateless_cartpole_ppo.py (hunk 4 above)
Comment (sven1977) [reply rllib/tuned_examples/ppo/multi_agent_stateless_cartpole_ppo.py:40]: correct.

**Review by simonsays1980 — APPROVED** [review]
LGTM.

File: rllib/examples/rl_modules/classes/modelv2_to_rlm.py (hunk 2 above)
Comment (sven1977) [reply rllib/examples/rl_modules/classes/modelv2_to_rlm.py:55]: I have a LSTM example script coming up in another PR (the one that adds the wrapper "from config").

**Pull request merged**